src/github.com/couchbase/sync_gateway/base/bucket.go

```diff
@@ -362,7 +363,12 @@ func GetCouchbaseBucket(spec BucketSpec, callback sgbucket.BucketNotifyFn) (buck
 func GetBucket(spec BucketSpec, callback sgbucket.BucketNotifyFn) (bucket Bucket, err error) {
 	if isWalrus, _ := regexp.MatchString(`^(walrus:|file:|/|\.)`, spec.Server); isWalrus {
 		Logf("Opening Walrus database %s on <%s>", spec.BucketName, spec.Server)
-		sgbucket.Logging = LogKeys["Walrus"]
+		if LogKeys["Walrus"] {
+			atomic.StoreUint32(&sgbucket.Logging, 1)
```

Comment:
As mentioned on the sg-bucket PR, I think that sgbucket consumers shouldn't need to know about the internal synchronization of the logging property - should push this handling down to sg-bucket.